### PR TITLE
fix: handle Objective-C exceptions in `StarlingAudioPlayer.play(_:identifier:)`

### PR DESCRIPTION
**Pull request by agurtovoy**
See https://github.com/matthewreagan/Starling/issues/7

### CODE REVIEW

File: Starling.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,6 +7,8 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		63B12B652925AA8A00F3251E /* ExceptionCatcher in Frameworks */ = {isa = PBXBuildFile; productRef = 63B12B642925AA8A00F3251E /* ExceptionCatcher */; };
```
Comment (agurtovoy) [inline Starling.xcodeproj/project.pbxproj:10]: Adding ExceptionCatcher dependency to the demo project.

File: Package.swift [hunk 2]
```diff
@@ -16,12 +16,15 @@ let package = Package(
...
+    dependencies: [
+        .package(url: "https://github.com/sindresorhus/ExceptionCatcher.git", from: "2.0.1")        
+    ],
     targets: [
         // Targets are the basic building blocks of a package. A target can define a module or a test suite.
         // Targets can depend on other targets in this package, and on products in packages this package depends on.
         .target(
             name: "Starling",
-            dependencies: [],
+            dependencies: ["ExceptionCatcher"],
```
Comment (agurtovoy) [inline Package.swift:27]: Adding ExceptionCatcher dependency to `Starling` package.

File: Starling/Starling.swift [hunk 3]
```diff
@@ -136,7 +138,11 @@ public class Starling {
             guard let player = firstAvailablePlayer() else { return }
 
             objc_sync_enter(players)
-            player.play(audio, identifier: sound)
+            do {
+                try player.play(audio, identifier: sound)
```
Comment (agurtovoy) [inline Starling/Starling.swift:142]: `player.play` can throw now.

File: Starling/Starling.swift [hunk 4]
```diff
@@ -291,13 +297,18 @@ private class StarlingAudioPlayer {
...
         node.scheduleFile(file, at: nil, completionCallbackType: .dataPlayedBack) {
             [weak self] callbackType in
             self?.didCompletePlayback(for: identifier)
         }
         state = PlayerState(sound: identifier, status: .playing)
-        node.play()
+        do {
+            try ExceptionCatcher.catch { node.play() }
+        } catch {
+            state = .idle()
```
Comment (agurtovoy) [inline Starling/Starling.swift:309]: Make sure `state` doesn't get stuck on `playing` if `node.play()` fails.

**Comment by matthewreagan** [comment]
Thanks @agurtovoy

**Pull request merged**